libs/api/books/src/lib/reading-list.service.ts

```diff
@@ -28,4 +28,14 @@ export class ReadingListService {
       return list.filter(x => x.bookId !== id);
     });
   }
+
+  async updateBook(id: string): Promise<void> {
+    this.storage.update(list => {
+      const bookIndex = list.findIndex(x => x.bookId === id);
+      const readDate = new Date().toISOString();
+      list[bookIndex].finished = true;
```

Comment:
I would check for `-1` here. Just to be safe

Reply:
Just added a check to validate that there was a match on the bookId